packages/i/imagemagick/xmake.lua

```diff
@@ -0,0 +1,97 @@
+    add_urls("https://download.imagemagick.org/ImageMagick/download/ImageMagick-$(version).tar.gz")
+    add_versions("7.0.11-13", "6c162d7cbd7b80968a6d083d39eb18d9c9bbd49f500f7f49c9a5dcc0fc36a03b")
+    add_deps("pkg-config")
+    add_configs("bzlib", {description = "Enable bzip2 support.", default = false, type = "boolean"})
+    add_configs("exr", {description = "Enable exr support.", default = false, type = "boolean"})
+    add_configs("fftw", {description = "Enable fftw support.", default = false, type = "boolean"})
+    add_configs("fontconfig", {description = "Enable fontconfig support.", default = false, type = "boolean"})
+    add_configs("freetype", {description = "Enable freetype support.", default = false, type = "boolean"})
+    add_configs("jpeg", {description = "Enable jpeg support through libjpeg.", default = true, type = "boolean"})
+    add_configs("lzma", {description = "Enable LZMA support.", default = false, type = "boolean"})
+    add_configs("openjpeg", {description = "Enable jpeg support through openjpeg.", default = false, type = "boolean"})
+    add_configs("png", {description = "Enable png support.", default = true, type = "boolean"})
+    add_configs("raw", {description = "Enable raw image support.", default = false, type = "boolean"})
+    add_configs("tiff", {description = "Enable tiff support.", default = false, type = "boolean"})
+    add_configs("threads", {description = "Enable threading support.", default = false})
+    add_configs("xml", {description = "Enable XML support.", default = false, type = "boolean"})
+    add_configs("webp", {description = "Enable webp support.", default = false, type = "boolean"})
+    add_includedirs("include/ImageMagick-7/")
+    add_links("MagickWand-7.Q16", "MagickCore-7.Q16", "Magick++-7.Q16")
+
+    on_load(function(package)
+        local configdeps = {bzlib      = "bzip2",
+                            exr        = "openexr",
+                            ffmpeg     = "ffmpeg",
+                            fontconfig = "fontconfig",
+                            freetype   = "freetype",
+                            fftw       = "fftw",
+                            jpeg       = "libjpeg",
+                            lzma       = "lzma",
+                            openjpeg   = "openjpeg",
+                            png        = "libpng",
+                            raw        = "libraw",
+                            tiff       = "libtiff",
+                            xml        = "libxml2",
+                            webp       = "libwebp"}
+        for name, dep in pairs(configdeps) do
+            if package:config(name) then
+                package:add("deps", dep)
+                if name == "tiff" then
+                    package:add("deps", "zlib")
+                end
+            end
+        end
+        if package:config("threads") and is_plat("linux") then
+            add_syslinks("pthread")
```

Comment:
package:add("syslinks", "pthread")

add_syslinks is ont the API of description scope, not in script scope.

Reply:
I can't use package outside of the script scope though?
When I do it results in:
```
error: ./packages/i/imagemagick/xmake.lua:25: attempt to index global 'package' (a function value)
stack traceback:
    [./packages/i/imagemagick/xmake.lua:25]: in main chunk
```
Lines 22-29 for context are:
```lua
    add_deps("pkg-config")
    add_includedirs("include/ImageMagick-7/")
    add_links("MagickWand-7.Q16", "MagickCore-7.Q16", "Magick++-7.Q16")
    if package:config("threads") and package:is_plat("linux") then
        add_syslinks("pthread")
    end

    on_load(function(package)
```

Reply:
no, please use `package:add("syslinks", "pthread")` if in on_load script scope